tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/upgrade/PreviousLiferayLocationValidationService.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+package com.liferay.ide.project.ui.upgrade;
+
+import java.io.File;
+import java.io.FilenameFilter;
+
+import org.eclipse.sapphire.modeling.Path;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.platform.StatusBridge;
+import org.eclipse.sapphire.services.ValidationService;
+
+import com.liferay.ide.project.core.ProjectCore;
+
+/**
+ * @author Lovett Li
+ */
+public class PreviousLiferayLocationValidationService extends ValidationService
+{
+    private final String[] PROPERTIES_FILENAME_PATTERNS = {
+        "portal-.*\\.properties",
+        "system-ext\\.properties",
+    };
+
+    @Override
+    protected Status compute()
```

Comment:
This should simply use the IRuntime#validate() class to perform validation.  This should not validate in any other way.  Ask Simon for how to perform this validation using the ServerCore plugin.